prow/pod-utils/gcs/upload.go

```diff
@@ -35,29 +35,44 @@ type UploadFunc func(obj *storage.ObjectHandle) error
 // Upload uploads all of the data in the
 // uploadTargets map to GCS in parallel. The map is
 // keyed on GCS path under the bucket
+// 3 attempts will be made to have an error free upload
 func Upload(bucket *storage.BucketHandle, uploadTargets map[string]UploadFunc) error {
+
 	errCh := make(chan error, len(uploadTargets))
 	group := &sync.WaitGroup{}
 	group.Add(len(uploadTargets))
+
 	for dest, upload := range uploadTargets {
+
 		obj := bucket.Object(dest)
 		logrus.WithField("dest", dest).Info("Queued for upload")
 		go func(f UploadFunc, obj *storage.ObjectHandle, name string) {
 			defer group.Done()
-			if err := f(obj); err != nil {
-				errCh <- err
+			retries := 3
+			for i := 1; i <= retries; i++ {
+				err := f(obj);
+				if err == nil {
+					break
+				} else if i == retries {
+					errCh <- err
+				}
+				logrus.WithField("dest", name).Info("Retrying upload")
```

Comment:
Can you include the error? `logrus.WithError(err).WithField(d,n).Info()`

Reply:
thanks for all the help and suggestions @fejta 

btw, I saw some tests failed, but the only one I could figure out was some code style issues. I
changed that, but wondering how to figure out the other failures. One failure log had this:

```
fatal: unable to access 'https://github.com/kubernetes/test-infra.git/': Could not resolve host: github.com
# Error: exit status 128
```

another like this [one](https://gubernator.k8s.io/build/kubernetes-jenkins/pr-logs/pull/test-infra/11133/pull-test-infra-bazel/24395) said, unable to load results.

do you know what I can do about those, or just wait for the tests to run again now that I've pushed an update?